### PR TITLE
[candi] add modulesoruce custom ca on nodes

### LINKED ISSUE #6222: Modules sources with custom CA does not work properly
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

main

### Expected Behavior

When we deploy module source with custom ca pods of downloaded module should start

### Actual Behavior

kubelet cannot download images from module source due to absence of custom ca on node.

### Steps To Reproduce

1. Create module source with custom CA.
2. Enable module
3. Check pods

### Additional Information

```
 Warning  Failed     9s               kubelet            Failed to pull image "1.2.3.4:8443/admin/deckhouse/fe/modules/deckhouse-admin@sha256:2f563c7c3d29b1f0d68fbaa66eef743ab255f4cbbcbcbe48ba59208911d10bc8": rpc error: code = Unknown desc = failed to pull and unpack image "1.2.3.4:8443/admin/deckhouse/fe/modules/deckhouse-admin@sha256:2f563c7c3d29b1f0d68fbaa66eef743ab255f4cbbcbcbe48ba59208911d10bc8": failed to resolve reference "1.2.3.4:8443/admin/deckhouse/fe/modules/deckhouse-admin@sha256:2f563c7c3d29b1f0d68fbaa66eef743ab255f4cbbcbcbe48ba59208911d10bc8": failed to do request: Head "https://1.2.3.4:8443/v2/admin/deckhouse/fe/modules/deckhouse-admin/manifests/sha256:2f563c7c3d29b1f0d68fbaa66eef743ab255f4cbbcbcbe48ba59208911d10bc8": x509: certificate signed by unknown authority
```

ModuleSource
```
apiVersion: deckhouse.io/v1alpha1
kind: ModuleSource
metadata:
  name: deckhouse
spec:
  registry:
    ca: |
      -----BEGIN CERTIFICATE-----
      MIIDXzCCAkegAwIBAgIULZtxHkic5sREb0xgRp54AzTi4xQwDQYJKoZIhvcNAQEL
      BQAwPzELMAkGA1UEBhMCUlUxDzANBgNVBAgMBk1vc2NvdzEPMA0GA1UEBwwGTW9z
      Y293MQ4wDAYDVQQKDAVmbGFudDAeFw0yMzEwMTIxMTU5NDZaFw0yNjA4MDExMTU5
      NDZaMD8xCzAJBgNVBAYTAlJVMQ8wDQYDVQQIDAZNb3Njb3cxDzANBgNVBAcMBk1v
      c2NvdzEOMAwGA1UECgwFZmxhbnQwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK
      AoIBAQDRUqFM+Hm/y3cYnybmtaghzRywXmtqCH8mp8ERu4xsqdI0Tn4sI+SGU065
      GfCZ13KrssfwuJR3eDguDw3XQs5IXxtvBDvy2SUgRMFJ0KTtSPIIkpm6MxXhRlAH
      DFxhslJJRe5V8Y9KRCohlUGDb9BrXS+ZgOW1C0la2hS8j+Qle7aRrRUaumj6rILl
      Rw+nvH89mfHzD6rwQuSsP+Vwq2wyYExoxXHhvi8Uk3XwNZvyt3u0Lq8IDyAJLBHw
      8UqBnI4YSfjEZC37PpVCN2gjgJFytYlvapD+gZfOLlrNXQiX7kVSOlsi+TCYmf07
      QrwNZk+lpwH7lGu3g50nwx6SXq+bAgMBAAGjUzBRMB0GA1UdDgQWBBR+DOxS4nOb
      Kg7kEFjGZJdFREJdKjAfBgNVHSMEGDAWgBR+DOxS4nObKg7kEFjGZJdFREJdKjAP
      BgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQCxgPP6PGSJEol2DbHn
      KJArj2UzmhsJP2XnOaIhgAO9IUEfC+vYRRcaDHnkkXXY2yd02FNK5Gmd1fUtWD8A
      QwqKLTWYlpVF8U13gdtZvr/tYS4ESfqLkCeH2Ncn2CgUxL1yV4GWALTsP4aoKjT2
      MLOL48Rv7vZB/lD5Dbc6bNHrc6POgFSuygNk1wTSGv+p2H6VnHpTTPxevAcgnpYC
      84aBtIE0Bfqq//mZzAb22teV/86meBw8ccO9IKK7vnNqR8htp6s04hNUENIW7tej
      su55hroS/bkn/ETWWclPJHEDSZ5n7V85Q+p7HXfUiD4LSc1JmafZwnpsdnDMWZkI
      yf3y
      -----END CERTIFICATE-----
    dockerCfg: __dockercfgjson__
    repo: 1.2.3.4:8443/admin/deckhouse/fe/modules
  releaseChannel: alpha
status:
  availableModules:
  - deckhouse-admin
  message: ""
  moduleErrors: []
  modulesCount: 1
  syncTime: "2023-10-12T18:21:00.137660207Z"
```

### Logs

_No response_

### PR DESCRIPTION
**Pull request by sprait**
## Description

Add custom CA from ModuleSource resource to node and configure containerd for using them. This is required to download module images from a private registries that uses its own CA.

## Why do we need it, and what problem does it solve?

Allow pull module images from registry, that uses its own certificate authority.
Closes #6222

## What is the expected result?

```
[root@dev-master-0 ms]# kubectl apply -f - <<EOF
apiVersion: deckhouse.io/v1alpha1
kind: ModuleSource
metadata:
  name: mysource
spec:
  registry:
    ca: |
      -----BEGIN CERTIFICATE-----
      MIIFtjCCA56gAwIBAgIUA3VmpztoKAKHlw6cEGSm6QxUfxMwDQYJKoZIhvcNAQEF
      BQAweDELMAkGA1UEBhMCWFgxDDAKBgNVBAgMA04vQTEMMAoGA1UEBwwDTi9BMSAw
      HgYDVQQKDBdTZWxmLXNpZ25lZCBjZXJ0aWZpY2F0ZTErMCkGA1UEAwwiMTIwLjAu
      MC4xOiBTZWxmLXNpZ25lZCBjZXJ0aWZpY2F0ZTAeFw0yNDA3MTAxMDA0NTNaFw0y
      NTA3MTAxMDA0NTNaMHgxCzAJBgNVBAYTAlhYMQwwCgYDVQQIDANOL0ExDDAKBgNV
      BAcMA04vQTEgMB4GA1UECgwXU2VsZi1zaWduZWQgY2VydGlmaWNhdGUxKzApBgNV
      BAMMIjEyMC4wLjAuMTogU2VsZi1zaWduZWQgY2VydGlmaWNhdGUwggIiMA0GCSqG
      SIb3DQEBAQUAA4ICDwAwggIKAoICAQCcA7qAP7oSQeZI06dwk7fgcZk3EleArnrO
      DeDl7x4kYoWPW4y5OouD/tLni4qdOrEuybrh0mirJz40o5EX5cTyjSs0AbX+VN3h
      DT78PuSeArvo87E65B0tQlytLm/CayeH0+O8z7H0ifCSUr1aNAlB+t/931BMttxJ
      4qDc+HT5gZxe0tBjVYFqjUkCxESxZDpJMGKCL6FMdlhOhcfHRd+kjbPS9oiKm7bb
      VzmYyCrGfFaekMtc3ZNy9AZB7LbhK+yolfdNzSF4ac5l5QSfsqao2as9oOfZsUob
      k6sIv183oWKgZFYEYLL3sSgArDC+5Xo+CrQyCaaW5GyjYGxhs1qwIHl3A7PPdt8Q
      NjwR4pit5wUBDAmI3id9htfH2U+7y7JLSkP8uXTa5ovJMRsFT4t/+cWjXs9anJ9C
      Cdcco/OouNdaMxY9dF1HtUeoqJLM7bVxsHER8iQd2QwQKbXquPA4ajx4Mml2hP5+
      /BXPAqynNf8EcU2A8qez6ICCTZdH2A3S/6Acwq5OJxGDlRuGwdXEtA8bN4Mv1fv9
      hgyydXbnG9/Se0oPJZWqr2OxVS4jVH4SXjsu+tnnBOG0YEu7rGH2AK//qkd+R7T2
      fx50+Ew3KFbpuezWjcqJw56kCzszv3LYLUys3Lr+xC2KhRJst1wqzeyV972nBaTX
      vGgVzkHzCwIDAQABozgwNjAVBgNVHREEDjAMhwTAqMdchwRf2UTgMB0GA1UdDgQW
      BBRPGOUnx1vOMgJ1UF3/028O+r4y5zANBgkqhkiG9w0BAQUFAAOCAgEADhnLrJB5
      8QmRfAmaaI4pyz5MWq8Eda6BsZeNuOLa8FtRGB4BG5vZ8Ye4yeOQiIZ8GuT4XDrX
      8eIkZ3KE23eLZiN9jJ589Ciod0JaXVBKi+QC0v9d0CPCZmoCme9ZBqq5m9vrJ7yN
      hkvltFzWhl+al9esa5vcu8pDEbo9N7m4tVLMZpMDJ5xIWmfgvkHWaDR2nxFHBJ6E
      RhppGsXEeX3iJ0EWaEJJxGYDMR0MGBRCEYoMCvYyodjLJF2o9GatTCNmVNo0sSQ7
      pgY5pAhDC2XC1YvC4fLX00D5uOXLhYvdEXBNI/qGtVH2p7oQpU8tkUzypJe32sFC
      u1M1liHt0Pf7I850c1AJ5blGwKrW4SuTqB91tiH5T5XWcWjMTKhnbJgOLqbedlx+
      YEMoZF4nrctIRQ7+uLttJTFiyiHzk+f5yTLrMKcUGckFDG/iBhIqfhOWsUkBMGxB
      oCFJlLzLSIw+wNVwlICJp70ecsBPezj9f6qDUuHX61VQpkyPfH2y/56dIhscPxsB
      aq650hIAVv22yJDDWrpgr0hFXg2QQwbz08C4aSWB1tIPDztm/F0S4WyHV3je61BR
      fHj1jBdLcj/hkdG8wHNx1mPvP5/Bbe17gdrED0sAAAdzx2kVJd/xQemExIcqYKBy
      L7rLB/TYLHaeIcbf1NF89Dr45TNixBGnZb0=
      -----END CERTIFICATE-----
    dockerCfg: $DOCKERCFG
    repo: 192.168.199.92:5000/deckhouse-modules
    scheme: HTTPS
  releaseChannel: ""
EOF
modulesource.deckhouse.io/mysource created
[root@dev-master-0 ms]#
[root@dev-master-0 ms]# kubectl apply -f -  <<EOF
---
apiVersion: deckhouse.io/v1alpha1
kind: ModuleConfig
metadata:
  name: kubectl-plugins
spec:
  enabled: true
  settings:
    defaultPlugins: true
  version: 1
EOF
moduleconfig.deckhouse.io/kubectl-plugins configured

```

Behavior before:
```
  Warning  Failed     4s    kubelet            Failed to pull image "192.168.199.92:5000/deckhouse-modules/kubectl-plugins@sha256:490c217d9800cdd41c68fe7c9b21c0d24bdbc459292b791c6bc8c3bb3114f038": rpc error: code = Unknown desc = failed to pull and unpack image "192.168.199.92:5000/deckhouse-modules/kubectl-plugins@sha256:490c217d9800cdd41c68fe7c9b21c0d24bdbc459292b791c6bc8c3bb3114f038": failed to resolve reference "192.168.199.92:5000/deckhouse-modules/kubectl-plugins@sha256:490c217d9800cdd41c68fe7c9b21c0d24bdbc459292b791c6bc8c3bb3114f038": failed to do request: Head "https://192.168.199.92:5000/v2/deckhouse-modules/kubectl-plugins/manifests/sha256:490c217d9800cdd41c68fe7c9b21c0d24bdbc459292b791c6bc8c3bb3114f038": tls: failed to verify certificate: x509: certificate signed by unknown authority
```

Behavior after:
```
  Normal   Pulling    52s                    kubelet            Pulling image "192.168.199.92:5000/deckhouse-modules/kubectl-plugins@sha256:490c217d9800cdd41c68fe7c9b21c0d24bdbc459292b791c6bc8c3bb3114f038"
  Normal   Pulled     49s                    kubelet            Successfully pulled image "192.168.199.92:5000/deckhouse-modules/kubectl-plugins@sha256:490c217d9800cdd41c68fe7c9b21c0d24bdbc459292b791c6bc8c3bb3114f038" in 2.171321696s (2.171343686s including waiting)
```

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: add modulesoruce custom ca on nodes
impact: If cluster has moduleSource resources with custom CA, containerd will be restarted on all nodes in cluster.
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by sprait** [comment]
/e2e/abort static;Static;containerd;1.27 9906949518 9906949518-1-con-1-27 /sys/deckhouse-oss/install:pr9000 astra@192.168.199.95

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/openstack'...

File: deckhouse-controller/crds/module-source.yaml [hunk 1]
```diff
@@ -68,7 +68,7 @@ spec:
                     ca:
                       type: string
                       description: |
-                        Root CA certificate (PEM format) to validate the registry’s HTTPS certificate (if self-signed certificates are used).
+                        Root CA certificate (PEM format) to validate the registry’s HTTPS certificate (if self-signed certificates are used). **Creating a ModuleSource resource with the CA certificate spec will cause the container to restart on all nodes.**
```
Comment (z9r5) [inline deckhouse-controller/crds/module-source.yaml:71]:
```suggestion
                        Root CA certificate (PEM format) to validate the registry’s HTTPS certificate (if self-signed certificates are used).
                        > Creating a ModuleSource resource with the CA certificate spec will cause the container to restart on all nodes.
```

File: deckhouse-controller/crds/doc-ru-module-source.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ spec:
                       description: Строка с токеном доступа к container registry в Base64.
                     ca:
                       description: |
-                        Корневой сертификат (В формате PEM), которым можно проверить сертификат registry при работе по HTTPS (если registry использует самоподписанные SSL-сертификаты).
+                        Корневой сертификат (В формате PEM), которым можно проверить сертификат registry при работе по HTTPS (если registry использует самоподписанные SSL-сертификаты). **Создание ресурса ModuleSource с задействованным параметром корневого сертификата, приведет к перезапуску containerd на всех нодах кластера.**
```
Comment (z9r5) [inline deckhouse-controller/crds/doc-ru-module-source.yaml:26]:
```suggestion
                        Корневой сертификат (В формате PEM), которым можно проверить сертификат registry при работе по HTTPS (если registry использует самоподписанные SSL-сертификаты).
                        > Создание ресурса ModuleSource с указанием корневого сертификата, приведет к перезапуску containerd на всех узлах кластера.
```

**Review by z9r5 — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Pull request merged**